### ISSUE TITLE
Add strokeDasharray to legend

### ISSUE THREAD

**Issue by remojansen** [recharts/recharts#970]
## Do you want to request a feature or report a bug?
New Feature

## What is the current behavior?
When displaying two lines with the same color like for example:

```ts
<Line type="monotone" dataKey="pv" stroke="#8884d8">
<Line type="monotone" dataKey="uv" stroke="#8884d8" strokeDasharray="3 4 5 2"/>
```

One with `strokeDasharray` and one without, the legend is not very useful:

![screenshot from 2017-10-25 10-56-59](https://user-images.githubusercontent.com/10656223/31993158-3e06c606-b974-11e7-995e-b54e1a79f6dd.png)

## What is the expected behavior?
The legend should allow us to identify which of the lines in the chart is the one with `strokeDasharray`.

# Possible implementation
I would like to send a PR to fix this but I need some help to find the right solution.

At the moment the `iconType` is a property of `Legend`. 

I was thinking that:

- If the `iconType` is set to `line` we can display a dashed line if `strokeDasharray` is used. 

- If the `iconType` is set to `square`, `rect`, `circle`, `cross`, `diamond`, `star`, `triangle` or `wye` we will not be able to differentiate between items with and without `strokeDasharray`.

@xile611 Can you please share your thoughts? Thanks!

**Comment by remojansen** [comment]
I have created a PR :point_right: #985

**Comment by xile611** [comment]
This feature has been released in 1.0.0-beta.5.

**Comment by remojansen** [comment]
Thanks a lot :+1:
